ghost/member-attribution/lib/referrer-translator.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * Translates referrer data into source and medium
+ */
+class ReferrerTranslator {
+    /**
+     *
+     * @param {Object} deps
+     * @param {Object} deps.urlUtils
+     */
+    constructor({urlUtils}) {
+        this.urlUtils = urlUtils;
+    }
+
+    getAttributionData({referrerUrl, referrerSource, referrerMedium}) {
+        return {
```

Comment:
Does it make more sense to return `null` here when there is no source or medium?

That way this class is solely responsible for generating referrer data, including what is/isn't a valid referrer

Reply:
Yeah that makes sense, though I am wondering if this should change completely where instead of calculating referrer for each individual history item, we might as well pass in the whole history here so it can have some smart calculations on how to derive the `Source` and `Medium` based on different history items. Gonna try that, otherwise switch to returning `null` if it keeps the current check for individual history item

Reply:
@allouis Updated the logic here to pass in the `history` directly, as the logic to smartly calculate referrer information needs access to whole history so it can be looped over for different cases. Also now returns `null` instead of object with nulls when no referrer information can be created, lmk if this makes sense.